### PR TITLE
fix(snapshot): reject multiple `toMatchInlineSnapshot` updates at the same location

### PR DESCRIPTION
**Pull request by hi-ogawa**
### Description

- Closes https://github.com/vitest-dev/vitest/issues/6327

There are some odd cases not handled well, but I think this will catch common cases and notify users effectively with an understandable error message. I tested on Jest and the behaviors of the test cases added here are same.

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [x] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [x] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

**Review by sheremet-va — APPROVED** [review]

File: test/cli/fixtures/fails/inline-snapshop-inside-loop-update-none.test.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+import {test, expect} from "vitest";
+
+// this test causes infinite re-run when --watch and --update
+// since snapshot update switches between "foo" and "bar" forever.
```
Comment (sheremet-va) [inline test/cli/fixtures/fails/inline-snapshop-inside-loop-update-none.test.ts:4]: should we do something about this?

Comment (hi-ogawa) [reply test/cli/fixtures/fails/inline-snapshop-inside-loop-update-none.test.ts:4]: Yeah, this is not ideal but I couldn't come up with a quick solution from what's available currently.
I think this is rather an edge case compared to OP's case, so we could postpone a proper fix later until someone comes across this. Btw, this behavior is same for current Vitest so that's unchanged. Jest also ends up with infinite re-run for this case.

Mentioning an idea for a proper fix, it would probably require keeping track of all `toMatchInlineSnapshot` call locations, not only failed ones. Currently `parseErrorStacktrace` is not used when `toMatchInlineSnapshot` succeeds, so I need to move up some code from `SnapshotState._addSnapshot` and also adjust states probably.

**Comment by odinho** [comment]
Ouch, we were relying on this, to make sure 3 different markdown parsers were giving the exact same end-result :P

Yeah, it was maybe abusing it. :shrug: :)
